docs/reference/modules/uploadfs.md

```diff
@@ -0,0 +1,359 @@
+
+#### Example
+```javascript
+async handleFile( file, uploadfsPath) {  
+  const copyIn = require('util').promisify(self.uploadfs.copyIn);
+  try {
+    await copyIn(file, uploadfsPath)
+  } catch (e) {
+    console.error(e);
+};  
+```
+
+### `copyOut(localPath, path, options, callback)`
+Much like the `copyIn` method, the `copyOut` method takes a `localPath` and a uploadfs `path` and copies the file back from uploadfs to the local file system. This is used when an existing asset needs to be updated, such as supplying additional cropping or other image processing. It should be used sparingly because it can result in poor storage service performance. Like the `copyIn` method, wrapping `copyOut` in `promisify` is often useful.
+
+#### Example
+
+```javascript
+async modifyImage( file, uploadfsPath) {  
+  const copyOut = require('util').promisify(self.uploadfs.copyOut);
+  const copyIn = require('util').promisify(self.uploadfs.copyIn);
+  try {  
+    await copyOut(file, uploadfsPath);
+    await newImageCrop(file);
+    await copyIn(file, uploadfsPath);
+   } catch (err) {
+    console.error(err);
+  }
+};
+```
+
+### `migrateFromDisabledFileKey(callback)`
+The `migrateFromDisabledFileKey` function provides a way to switch any file previously disabled using the `disabledFileKey` option to use the `disable` method instead. Note: the `disabledFileKey` option should be removed before invoking this method.
+
+#### Example
+
+```javascript
+self.uploadfs.migrateFromDisabledFileKey(function(e) { … });
+```
+
+### `migrateToDisabledFileKey(callback)`
+The `migrateToDisabledFileKey` function provides a way to switch any file previously disabled using the`disable` method to use the `disabledFileKey` option.
+
+#### Example
+
+```javascript
+self.uploadfs.migrateToDisabledFileKey(function(e) { … });
+```
+
+## ENVIRONMENT variables
+
+### `APOS_S3_BUCKET`
+Apostrophe exposes a number of environmental variables for easily setting a number of options for the AWS S3 service allows. The 'APOS_S3_BUCKET' must be set to the bucket name to set the account options in this way. The other possible environment variables are `APOS_S3_ENDPOINT`, `APOS_S3_SECRET`, `APOS_S3_KEY`, and `APOS_S3_REGION`. The values of these variables will override any values passed in through the corresponding option. No other cconfiguration is required.
```

Comment:
Please document all of these individually under their own subheadings, this is the "front door" most people use in practice to configure this module in production. You will typically need all of BUCKET, SECRET, and KEY. If you are using official Amazon-provided S3 you'll need REGION. If you're using a third party provider you'll need ENDPOINT instead.

Reply:
cconfiguration -> configuration

Expand on this point. "The easiest way to configure this module for S3-compatible storage is to set the following environment variables. If they are present Apostrophe will automatically switch the storage strategy to `s3`."

I think you should move the environment variables section higher.

Reply:
Tried to clarify and moved the section before the methods.